---
title: 'Report 8B: Contact and expression'
author: "Cyril Matthey-Doret"
date: "19 novembre 2016"
output: pdf_document
---

#Introduction: 

```{r setup, echo=F}
#setwd("/home/cyril/Documents/Master/sem_1/First_step/data/")
#setwd("/Users/cmatthe5/Documents/First_step/data/")
setwd("/home/cyril/Documents/First_step/data/")

# Loading lincRNAs sets
ne_linc <- read.table("enhancer_bound/all_combinations/ne_linc_prb.bed")  # Overlap no enhancer marks, does not take promoters into account
e_linc <- read.table("enhancer_bound/all_combinations/e_linc_prb.bed")  # Overlap enhancer marks, does not take promoters into account
ne.p_linc <- read.table("enhancer_bound/all_combinations/ne.p_linc_prb.bed")  # Overlap promoter marks, but no enhancer marks
e.np_linc <- read.table("enhancer_bound/all_combinations/e.np_linc_prb.bed")  # Overlap enhancer marks, but no promoter marks
e.p_linc <- read.table("enhancer_bound/all_combinations/e.p_linc_prb.bed")  # Overlap promoter marks, and enhancer marks
ne.np_linc <- read.table("enhancer_bound/all_combinations/ne.np_linc_prb.bed")  # Overlap neither promoter marks, nor enhancer marks

# Loading pcgenes sets
ne_pc <- read.table("enhancer_bound/all_combinations/ne_pc_prb.bed")  # Overlap no enhancer marks, does not take promoters into account
e_pc <- read.table("enhancer_bound/all_combinations/e_pc_prb.bed")  # Overlap enhancer marks, does not take promoters into account
ne.p_pc <- read.table("enhancer_bound/all_combinations/ne.p_pc_prb.bed")  # Overlap promoter marks, but no enhancer marks
e.np_pc <- read.table("enhancer_bound/all_combinations/e.np_pc_prb.bed")  # Overlap enhancer marks, but no promoter marks
e.p_pc <- read.table("enhancer_bound/all_combinations/e.p_pc_prb.bed")  # Overlap promoter marks, and enhancer marks
ne.np_pc <- read.table("enhancer_bound/all_combinations/ne.np_pc_prb.bed")  # Overlap neither promoter marks, nor enhancer marks

#adding colnames
colnames(ne_linc)= colnames(e_linc)=colnames(ne.p_linc)=colnames(e.np_linc)=colnames(e.p_linc)=
  colnames(ne.np_linc)=colnames(ne_pc)=colnames(e_pc)=colnames(ne.p_pc)=colnames(e.np_pc)=colnames(e.p_pc)=
  colnames(ne.np_pc) <-c("chr", "start", "end", "gene", "strand")


rawviz <- function(hic, expr, cell_line="", gtype=""){  # This function computes the correlation between expression and DNA contacts for a set of genes.
# Both arguments need to be a list of genes containing the gene ID and its median expression level/amount of DNA-DNA contact.
  merged <- merge(x = hic, y = expr, by = "gene", all = F)
  merged$expression <- as.numeric(merged$expression)
  merged$diam <- as.numeric(merged$diam)
  plot(merged$diam, merged$expression, xlim = c(0,200000), ylim = c(0,1000),
                xlab = "DNA-DNA interactions",ylab="Median expression levels", main=paste(cell_line,gtype,sep=": "),pch=1)
  text(x=100000,y=800, labels = paste0("Med exp: ", round(median(merged$expression),3)))
  text(x=100000,y=600, labels = paste0("Med int: ", round(median(merged$diam),3)))
  
}

contexpr <- function(hic, expr, cell_line="", gtype=""){  # This function computes the correlation between expression and DNA contacts for a set of genes.
# Both arguments need to be a list of genes containing the gene ID and its median expression level/amount of DNA-DNA contact.
  merged <- merge(x = hic, y = expr, by = "gene", all = F)
  merged$expression <- as.numeric(merged$expression)
  merged$diam <- as.numeric(merged$diam)
  results <- c(cor.test(merged$diam, merged$expression, method="spearman")$estimate, 
               cor.test(merged$diam, merged$expression, method="spearman")$p.value)
  plot(xlim=c(0,8),ylim=c(-3,4),log10(merged$diam), log10(merged$expression), pch=".", 
                xlab = "Log10 DNA-DNA interactions",ylab="Log10 median expression levels", main=paste(cell_line,gtype,sep=": "))
  text(x = c(6.5,6.5,6.5),y=c(3,2.5,2), labels=c(paste0("rho = ", round(results[1],3)), paste0("p-value = ", round(results[2],3)),
                                                 paste0("n = ", length(merged$diam))))
}

extract_int <- function(cset, gset){
  csubset <- cset[cset$gene %in% gset$gene,]
  csubset <- csubset[csubset$diam!=0,]
  return(csubset)
}

linclist <- list(e_linc,e.p_linc,e.np_linc,ne_linc,ne.p_linc,ne.np_linc)
pclist <- list(e_pc,e.p_pc,e.np_pc,ne_pc,ne.p_pc,ne.np_pc)
lincnames <- c("elincRNA","eplincRNA","enplincRNA","nelincRNA","neplincRNA","nenplincRNA")
pcnames <- c("ePCG","epPCG","enpPCG","nePCG","nepPCG","nenpPCG")

```

Here, I investigate the relationship between gene expression and DNA-DNA contact across 4 different cell lines (GM12878, K562, HUVEC and NHEK). This analysis is performed across all LCL expressed lincRNAs and protein-coding genes, categorized by according to their overlap with enhancers and promoters. Contact per gene was calculated in 2 ways: Gene versus chromosome conmputes all interactions between the gene and the chromosome, while gene versus TAD only takes interactions between the gene and the TAD it belongs to.

NOTE 1: At the moment I only have expression data for GM12878, but I will add the other cell lines later.
NOTE 2: Contact matrices for chromosome 9 for all cell lines have been normalised using SQRTVC instead of KR because the algorithm did not converge for chromosome 9 in K562 and vector was full of NAs.


#Results: 

## Overview of data:


A quick overview of the data shows that enhancer-associated genes have a lower expression and higher amounts of DNA contacts on average. Patterns in the data would be easier to detect after log transforming both variables.

###Gene versus chromosome:

```{r raw_visu_chrom_linc, echo=F,fig.height=7}

#setwd("/home/cyril/Documents/Master/sem_1/First_step/data/")
#setwd("/Users/cmatthe5/Documents/First_step/data/")
setwd("/home/cyril/Documents/First_step/data/")
par(mfrow=c(2,3))


for(c in c("GM12878")){
  linc_contact <- read.table(paste0("TAD_contact/byChrom/all.lincRNA.",c,".HiC.contact.txt"), header=T)
  linc_expr <- read.table("expression/LCL.lincRNA.expression.txt")
  colnames(linc_expr) <- c("gene", "expression")
  count <- 1
  for(var in linclist){
    expr <- linc_expr[linc_expr$gene %in% var$gene,]
    contact <- extract_int(linc_contact,var)
    rawviz(contact, expr,cell_line=c,gtype=lincnames[count])
    count <- count+1
  }
}


```

```{r raw_visu_chrom_pc, echo=F,fig.height=7}

setwd("/home/cyril/Documents/First_step/data/")
#setwd("/home/cyril/Documents/Master/sem_1/First_step/data/")
#setwd("/Users/cmatthe5/Documents/First_step/data/")
par(mfrow=c(2,3))


for(c in c("GM12878")){
  pc_contact <- read.table(paste0("TAD_contact/byChrom/all.pcgene.",c,".HiC.contact.txt"), header=T)
  pc_expr <- read.table("expression/LCL.pcgene.expression.txt")
  colnames(pc_expr) <- c("gene", "expression")
  count <- 1
  for(var in pclist){
    expr <- pc_expr[pc_expr$gene %in% var$gene,]
    contact <- extract_int(pc_contact,var)
    rawviz(contact, expr,cell_line=c,gtype=pcnames[count])
    count <- count+1
  }
}

```


###Gene versus TAD:

```{r raw_visu_TAD_linc, echo=F,fig.height=7}

setwd("/home/cyril/Documents/First_step/data/")
#setwd("/home/cyril/Documents/Master/sem_1/First_step/data/")
#setwd("/Users/cmatthe5/Documents/First_step/data/")
par(mfrow=c(2,3))

for(c in c("GM12878")){
  linc_contact <- read.table(paste0("TAD_contact/byTAD/all.lincRNA.",c,".HiC.contact.txt"), header=T)
  linc_expr <- read.table("expression/LCL.lincRNA.expression.txt")
  colnames(linc_expr) <- c("gene", "expression")
  count <- 1
  for(var in linclist){
    expr <- linc_expr[linc_expr$gene %in% var$gene,]
    contact <- extract_int(linc_contact,var)
    rawviz(contact, expr,cell_line=c,gtype=lincnames[count])
    count <- count+1
  }
}


```

```{r raw_visu_TAD_pc, echo=F,fig.height=7}

setwd("/home/cyril/Documents/First_step/data/")
#setwd("/home/cyril/Documents/Master/sem_1/First_step/data/")
#setwd("/Users/cmatthe5/Documents/First_step/data/")
par(mfrow=c(2,3))

for(c in c("GM12878")){
  pc_contact <- read.table(paste0("TAD_contact/byTAD/all.pcgene.",c,".HiC.contact.txt"), header=T)
  pc_expr <- read.table("expression/LCL.pcgene.expression.txt")
  colnames(pc_expr) <- c("gene", "expression")
  count <- 1
  for(var in pclist){
    expr <- pc_expr[pc_expr$gene %in% var$gene,]
    contact <- extract_int(pc_contact,var)
    rawviz(contact, expr,cell_line=c,gtype=pcnames[count])
    count <- count+1
  }
}

```

##Correlations:

Spearman correlation was used to detect non-linear relationship between DNA contact and expression. The data was log-transformed for visualization.

###Gene versus chromosome:

```{r correlation_chrom_linc, echo=F,fig.height=7, warning=F}

setwd("/home/cyril/Documents/First_step/data/")
#setwd("/home/cyril/Documents/Master/sem_1/First_step/data/")
#setwd("/Users/cmatthe5/Documents/First_step/data/")
par(mfrow=c(2,3))

for(c in c("GM12878")){
  linc_contact <- read.table(paste0("TAD_contact/byChrom/all.lincRNA.",c,".HiC.contact.txt"), header=T)
  linc_expr <- read.table("expression/LCL.lincRNA.expression.txt")
  colnames(linc_expr) <- c("gene", "expression")
  count <- 1
  for(var in linclist){
    expr <- linc_expr[linc_expr$gene %in% var$gene,]
    contact <- extract_int(linc_contact,var)
    contexpr(contact, expr,cell_line=c,gtype=lincnames[count])
    count <- count+1
  }
}

```

```{r correlation_chrom_pc, echo=F,fig.height=7, warning=F}

setwd("/home/cyril/Documents/First_step/data/")
#setwd("/home/cyril/Documents/Master/sem_1/First_step/data/")
#setwd("/Users/cmatthe5/Documents/First_step/data/")
par(mfrow=c(2,3))

for(c in c("GM12878")){
  pc_contact <- read.table(paste0("TAD_contact/byChrom/all.pcgene.",c,".HiC.contact.txt"), header=T)
  pc_expr <- read.table("expression/LCL.pcgene.expression.txt")
  colnames(pc_expr) <- c("gene", "expression")
  count <- 1
  for(var in pclist){
    expr <- pc_expr[pc_expr$gene %in% var$gene,]
    contact <- extract_int(pc_contact,var)
    contexpr(contact, expr,cell_line=c,gtype=pcnames[count])
    count <- count+1
  }
}

```



###Gene versus TAD:

```{r correlation_TAD_linc, echo=F,fig.height=7, warning=F}

setwd("/home/cyril/Documents/First_step/data/")
#setwd("/home/cyril/Documents/Master/sem_1/First_step/data/")
#setwd("/Users/cmatthe5/Documents/First_step/data/")
par(mfrow=c(2,3))

for(c in c("GM12878")){
  linc_contact <- read.table(paste0("TAD_contact/byTAD/all.lincRNA.",c,".HiC.contact.txt"), header=T)
  linc_expr <- read.table("expression/LCL.lincRNA.expression.txt")
  colnames(linc_expr) <- c("gene", "expression")
  count <- 1
  for(var in linclist){
    expr <- linc_expr[linc_expr$gene %in% var$gene,]
    contact <- extract_int(linc_contact,var)
    contexpr(contact, expr,cell_line=c,gtype=lincnames[count])
    count <- count+1
  }
}

```

```{r correlation_TAD_pc, echo=F,fig.height=7, warning=F}

setwd("/home/cyril/Documents/First_step/data/")
#setwd("/home/cyril/Documents/Master/sem_1/First_step/data/")
#setwd("/Users/cmatthe5/Documents/First_step/data/")
par(mfrow=c(2,3))

for(c in c("GM12878")){
  pc_contact <- read.table(paste0("TAD_contact/byTAD/all.pcgene.",c,".HiC.contact.txt"), header=T)
  pc_expr <- read.table("expression/LCL.pcgene.expression.txt")
  colnames(pc_expr) <- c("gene", "expression")
  count <- 1
  for(var in pclist){
    expr <- pc_expr[pc_expr$gene %in% var$gene,]
    contact <- extract_int(pc_contact,var)
    contexpr(contact, expr,cell_line=c,gtype=pcnames[count])
    count <- count+1
  }
}

```

#Conclusions: 

In general, there is a weak, non-linear relationship between expression and DNA-DNA contact. Genes with a lot of contact have very small expression levels, while those with little contact can have both low and high expression levels.

The correlation is weaker in elincRNAs than in nelincRNAs. This may simply be caused by the overall lower expression of enhancer-bound genes.
This low expression may be due to bidirectionally transcribed RNAs that are lowly expressed. It would thus be interesting to differentiate between unidirectional and bidirectional transcription.

The relationship observed between contact and expression can in fact not really be called a correlation, but there is still an interesting difference between the patterns observed in lincRNAs and protein-coding genes. Note the graphs has been rescaled, it appears that elinc RNAs are a subset of nelincRNAs and that these 2 groups do not present significant differences.
