---
title: 'Report 8: Contact and expression'
author: "Cyril Matthey-Doret"
date: "9 novembre 2016"
output: pdf_document
---

#Introduction: 

```{r setup, echo=F}
setwd("/home/cyril/Documents/Master/sem_1/First_step/data/")

elinc <- read.table("enhancer_bound/elinc_prb.bed")
nelinc <- read.table("enhancer_bound/nelinc_prb.bed")
colnames(elinc) = colnames(nelinc) <- c("chr", "start", "end", "gene", "strand")

epc <- read.table("enhancer_bound/epc_prb.bed")
nepc <- read.table("enhancer_bound/nepc_prb.bed")
colnames(epc) = colnames(nepc) <- c("chr", "start", "end", "gene", "strand")

contexpr <- function(hic, expr, cell_line="", gtype=""){  # This function computes the correlation between expression and DNA contacts for a set of genes.
# Both arguments need to be a list of genes containing the gene ID and its median expression level/amount of DNA-DNA contact.
  merged <- merge(x = hic, y = expr, by = "gene", all = F)
  results <- c(cor.test(merged$diam, merged$expression, method="spearman")$estimate, 
               cor.test(merged$diam, merged$expression, method="spearman")$p.value)
  smoothScatter(log10(merged$diam), log10(merged$expression), 
                xlab = "Log10 DNA-DNA interactions",ylab="Log10 median expression levels", main=paste(cell_line,gtype,sep=": "))
  text(x = c(6.5,6.5),y=c(1.5,1), labels=c(paste0("rho = ", round(results[1],3)), paste0("p-value = ", round(results[2],3))))
  
}


```

Here, I investigate the relationship between gene expression and DNA-DNA contact across 4 different cell lines (GM12878, K562, HUVEC and NHEK). This analysis is performed on elincRNAs, nelincRNAs, epcRNAs and nepcRNAs.

NOTE 1: At the moment I only have expression data for GM12878, but I will add the other cell lines later.
NOTE 2: Contact matrices for chromosome 9 for all cell lines have been normalised using SQRTVC instead of KR because the algorithm did not converge for K562 and vector was full of NAs.


#Results: 

## Overview of data:


A quick overview of the data shows that enhancer-associated genes have a lower expression and higher amounts of DNA contacts on average. Patterns in the data would be easier to detect after log transforming both variables.

```{r raw_visu, echo=F,fig.height=7}

rawviz <- function(hic, expr, cell_line="", gtype=""){  # This function computes the correlation between expression and DNA contacts for a set of genes.
# Both arguments need to be a list of genes containing the gene ID and its median expression level/amount of DNA-DNA contact.
  merged <- merge(x = hic, y = expr, by = "gene", all = F)
  plot(merged$diam, merged$expression, 
                xlab = "DNA-DNA interactions",ylab="Median expression levels", main=paste(cell_line,gtype,sep=": "),pch=16)
  text(x=max(merged$diam)/2,y=max(merged$expression)/2, labels = paste0("Median expression: ", round(median(merged$expression),3)))
  text(x=max(merged$diam)/2,y=max(merged$expression)/3, labels = paste0("Median interactions: ", round(median(merged$diam),3)))
  
}
setwd("/home/cyril/Documents/Master/sem_1/First_step/data/")
par(mfrow=c(2,2))
linc_contact <- read.table(paste0("TAD_contact/all.lincRNA.GM12878.HiC.contact.txt"), header=T)
linc_expr <- read.table("expression/LCL.lincRNA.expression.txt")
colnames(linc_expr) <- c("gene", "expression")
elinc_expr <- linc_expr[linc_expr$gene %in% elinc$gene,]
nelinc_expr <-linc_expr[linc_expr$gene %in% nelinc$gene,]
elinc_contact <- linc_contact[linc_contact$gene %in% elinc$gene,]
nelinc_contact <- linc_contact[linc_contact$gene %in% nelinc$gene,]
rawviz(elinc_contact, elinc_expr,cell_line="GM12878",gtype="elincRNA")
rawviz(nelinc_contact, nelinc_expr,cell_line="GM12878",gtype="nelincRNA")

epc <- read.table("enhancer_bound/epc_prb.bed")
nepc <- read.table("enhancer_bound/nepc_prb.bed")
colnames(epc) = colnames(nepc) <- c("chr", "start", "end", "gene", "strand")

pc_contact <- read.table(paste0("TAD_contact/all.pcgene.GM12878.HiC.contact.txt"), header=T)
pc_expr <- read.table("expression/LCL.pcgene.expression.txt")
colnames(pc_expr) <- c("gene", "expression")
epc_expr <- pc_expr[pc_expr$gene %in% epc$gene,]
nepc_expr <-pc_expr[pc_expr$gene %in% nepc$gene,]
epc_contact <- pc_contact[pc_contact$gene %in% epc$gene,]
nepc_contact <- pc_contact[pc_contact$gene %in% nepc$gene,]
rawviz(epc_contact, epc_expr,cell_line="GM12878",gtype="epc gene")
rawviz(nepc_contact, nepc_expr,cell_line="GM12878",gtype="nepc gene")

```

##Correlations:

Spearman correlation was used to detect non-linear relationship between DNA contact and expression. The data was log-transformed for visualization.

```{r tmp_results, echo=F, warning=F, fig.height=8, fig.width=8}

setwd("/home/cyril/Documents/Master/sem_1/First_step/data/")

par(mfrow=c(2,2))
#for(c in c("GM12878", "HUVEC", "K562", "NHEK")){
for(c in c("GM12878")){
  linc_contact <- read.table(paste0("TAD_contact/all.lincRNA.",c,".HiC.contact.txt"), header=T)
  linc_expr <- read.table("expression/LCL.lincRNA.expression.txt")
  colnames(linc_expr) <- c("gene", "expression")
  elinc_expr <- linc_expr[linc_expr$gene %in% elinc$gene,]
  nelinc_expr <-linc_expr[linc_expr$gene %in% nelinc$gene,]
  elinc_contact <- linc_contact[linc_contact$gene %in% elinc$gene,]
  nelinc_contact <- linc_contact[linc_contact$gene %in% nelinc$gene,]
  contexpr(elinc_contact, elinc_expr,cell_line=c,gtype="elincRNA")
  contexpr(nelinc_contact, nelinc_expr,cell_line=c,gtype="nelincRNA")
}

#================================
# Analysis for epc and nepc


#par(mfrow=c(2,2))
#for(c in c("GM12878", "HUVEC", "K562", "NHEK")){
for(c in c("GM12878")){
  pc_contact <- read.table(paste0("TAD_contact/all.pcgene.",c,".HiC.contact.txt"), header=T)
  pc_expr <- read.table("expression/LCL.pcgene.expression.txt")
  colnames(pc_expr) <- c("gene", "expression")
  epc_expr <- pc_expr[pc_expr$gene %in% epc$gene,]
  nepc_expr <-pc_expr[pc_expr$gene %in% nepc$gene,]
  epc_contact <- pc_contact[pc_contact$gene %in% epc$gene,]
  nepc_contact <- pc_contact[pc_contact$gene %in% nepc$gene,]
  contexpr(epc_contact, epc_expr,cell_line=c,gtype="epc gene")
  contexpr(nepc_contact, nepc_expr,cell_line=c,gtype="nepc gene")
}


```


#Conclusions: 

In general, there is a weak, non-linear relationship between expression and DNA-DNA contact. Genes with a lot of contact have very small expression levels, while those with little contact can have both low and high expression levels.

The correlation is weaker in elincRNAs than in nelincRNAs. This may simply be caused by the overall lower expression of enhancer-bound genes.
This low expression may be due to bidirectionally transcribed RNAs that are lowly expressed. It would thus be interesting to differentiate between unidirectional and bidirectional transcription.
