---
title: 'Report 7: GAT tweaks'
output: pdf_document
date: "4 November 2016"
---

##Introduction
Here I increased the sampling of GAT to 10000 instead of 1000 to give it more power. I also worked with 10 bins per TAD instead of 20. I only used segment overlap as a measure, not nucleotide. When using expressed lincRNA, intergenic space and all protein-coding genes as workspace, I had to change the --nbuckets parameter from the default of 100000 to 270000 so that nbuckets*bucket size was larger than my largest segment of 269000. This also slows down the computations. Finally, I split the binding sites for CTCF, SCM3 and RAD21 into 3 different analysis so that if the results are only positive for 1 of the proteins, it will not be hidden.

NOTE: Chromosome 9 Has been normalised using SQRTVC instead of KR because the algorithm did not converge for K562 and vector was full NA.

#Results

