---
title: 'Report 6: Redirection'
author: "Cyril Matthey-Doret"
date: "28 octobre 2016"
output: pdf_document
---

```{r setup, include=FALSE}
setwd("/Users/cmatthe5/Documents/First_step/data/")
whole_gat_nuc <- read.table("GAT/out/whole_nuc_gat_results.txt",header=T)
whole_gat_seg <- read.table("GAT/out/whole_seg_gat_results.txt",header=T)

bins2gene_nuc<-droplevels(whole_gat_nuc[whole_gat_nuc$segment=="short5bins",])
bins2gene_nuc$track <- as.character(bins2gene_nuc$track)
bins2gene_nuc$track <- factor(bins2gene_nuc$track,levels = c(1:20))

gene2bins_nuc<-droplevels(whole_gat_nuc[whole_gat_nuc$annotation=="short5bins",])
gene2bins_nuc$annottrack <- as.character(gene2bins_nuc$annottrack)
gene2bins_nuc$annottrack <- factor(gene2bins_nuc$annottrack,levels = c(1:20))

bs2gene_nuc<-droplevels(whole_gat_nuc[whole_gat_nuc$segment=="bindingsite",])
gene2bs_nuc<-droplevels(whole_gat_nuc[whole_gat_nuc$annotation=="bindingsite",])

nice_workspace <- c(wholegenome= "Whole genome",intergenic="Intergenic space",allpc="Protein-coding space", lincRNA="Expressed lincRNAs space")

bins2gene_seg<-droplevels(whole_gat_seg[whole_gat_seg$segment=="short5bins",])
bins2gene_seg$track <- as.character(bins2gene_seg$track)
bins2gene_seg$track <- factor(bins2gene_seg$track,levels = c(1:20))

gene2bins_seg<-droplevels(whole_gat_seg[whole_gat_seg$annotation=="short5bins",])
gene2bins_seg$annottrack <- as.character(gene2bins_seg$annottrack)
gene2bins_seg$annottrack <- factor(gene2bins_seg$annottrack,levels = c(1:20))

bs2gene_seg<-droplevels(whole_gat_seg[whole_gat_seg$segment=="bindingsite",])
gene2bs_seg<-droplevels(whole_gat_seg[whole_gat_seg$annotation=="bindingsite",])

library(ggplot2);library(gridExtra)
elements_names <- c(
  `pr` = "promoter",
  `prb` = "promoter + gene body",
  `elinc` = "elincRNA",
  `nelinc` = "non-elincRNA",
  `epc` = "epc gene",
  `nepc` = "non-epc gene"
)
```

#Introduction

The largely different gene length between protein-coding genes and lincRNAs made it more difficult than expected to find appropriate thresholds to call TAD-bound and non TAD-bound genes. The focus of the project has thus been redirected to investigate the role of enhancer-associated lincRNAs (elincRNAs) in the organization of TADs.

#Methods

   * __Definition of TAD-boundaries__: TAD boundaries are now only extended towards the inside of TADs at 5% of TAD length and when TAD boundaries overlap, the largest ones are removed. This prevent the boundaries of large encompassing TADs from masking those of smaller TADs inside.

   * __Definition of enhancer-associated genes__: LincRNAs and protein-coding genes whose promoter regions (or promoter region + gene body) was overlapping enhancer elements (defined by histone modification marks) were defined as elincRNAs or epc-genes. The others are considered non-enhancer-associated. Promoter regions have been defined as the region of 1kb around the TSS of each gene.
   
Gene type|region|Original set|Overlaps|enhancer-associated genes
---------|------|------------|--------|---
lincRNA|pro|2510|323|274 (`r round((100*274)/2510,1)`%)
lincRNA|pro+bod|2510|1544|611 (`r round((100*611)/2510,1)`%)
protein-coding|pro|14846|3411|3019 (`r round((100*3019)/14846,1)`%)
protein-coding|pro+bod|14846|34808|8486 (`r round((100*8486)/14846,1)`%)

   * __Generation of bins__: TADs have been split into bins of 5%. This was done after removing all large TADs overlapping smaller ones. Note the difference with the definition of TAD-boundaries, where only large overlapping boundaries have been removed.
   
   * __Testing enrichment at TAD boundaries__: The GAT framework is used to test if enhancer-bound lincRNA and protein-coding genes are enriched at TAD boundaries. For both gene types, it is done in two ways: testing enrichment of enhancer-bound and non-enhancer bound genes in the different bins, and testing enrichment of the different bins in enhancer-bound and non-enhancer bound genes.
   
   * __Transcription factors binding sites__: Chip-seq data for SMC1, SMC3, RAD21, SA1, SA2 and CTCF from the ENCODE website was used.


#Enrichment at TAD boundaries

Several GAT tests were performed using different combinations of segments, annotations and workspace. Both nucleotide overlap and number of overlapping segments were used as a  measure:

Workspace|Segment|Annotation
---------|-------|----------
Whole genome|[n]e(linc \| pc)pr[b]|(bins \| CTCF & cohesin bs)
Whole genome|(bins \| CTCF & cohesin bs)|[n]e(linc \| pc)pr[b]
intergenic|[n]elinc pr[b]|(bins \| CTCF & cohesin bs)
intergenic|(bins \| CTCF & cohesin bs)|[n]elinc pr[b]
all pc genes|[n]epc pr[b]|(bins \| CTCF & cohesin bs)
all pc genes|(bins \| CTCF & cohesin bs)|[n]epc pr[b]
expressed lincRNAs|elinc pr[b]|(bins \| CTCF & cohesin bs)
expressed lincRNAs|(bins \| CTCF & cohesin bs)|elinc pr[b]


where: 

bs = binding sites

e = enhancer bound 

ne = non-enhancer bound 

pr = promoter region

prb = promoter region and body

##Nucleotides overlap

###Analyses using TAD bins

```{r TAD_bins_enrich_nuc, echo=F, warning=F, fig.height=9,fig.width=9}

for(w in c("wholegenome","intergenic","allpc","lincRNA")){
  b2g <- ggplot(data=bins2gene_nuc[bins2gene_nuc$workspace==w,],aes(x=track,y=fold,fill=log10(qval)))+
    scale_fill_continuous(low="#66bb66",high="#bb6666",guide = "colourbar",name= "Log10\nq-value")+
    geom_bar(stat="identity")+theme_bw()+
    facet_grid(element~annotation,labeller = as_labeller(elements_names))+
    ggtitle(paste0(nice_workspace[w],": \nEnrichment of TAD bins in lincRNAs and pcgenes"))+
    xlab("TAD bins") + ylab("Fold enrichment")+
    theme(axis.text = element_text(size = rel(0.4)))
  
  g2b <- ggplot(data=gene2bins_nuc[gene2bins_nuc$workspace==w,],aes(x=annottrack,y=fold,fill=log10(qval)))+
    scale_fill_continuous(low="#66bb66",high="#bb6666",guide = "colourbar",name= "Log10\nq-value")+
    geom_bar(stat="identity")+theme_bw()+
    facet_grid(element~segment,labeller = as_labeller(elements_names))+
    ggtitle(paste0(nice_workspace[w],": \nEnrichment of lincRNAs and pc genes in TAD bins"))+
    xlab("TAD bins") + ylab("Fold enrichment")+
    theme(axis.text = element_text(size = rel(0.4)))
  grid.arrange(b2g,g2b)
}

```

###Analyses using CTCF/Cohesin binding sites

```{r bindingsite_enrich_nuc, echo=F, warning=F, fig.height=7,fig.width=9}

ggplot(data=bs2gene_nuc,aes(x=workspace,y=fold,fill=log10(qval)))+
  scale_fill_continuous(low="#66bb66",high="#bb6666",guide = "colourbar",name= "Log10 q-value")+
  geom_bar(stat="identity")+theme_bw()+
  facet_grid(element~annotation,labeller = as_labeller(elements_names))+
  ggtitle("Enrichment of CTCF/cohesin binding sites in lincRNAs and pcgenes")+
  xlab("Workspace") + ylab("Fold enrichment")+theme(axis.text.x=element_text(angle = -90, hjust = 0))

ggplot(data=gene2bs_nuc,aes(x=workspace,y=fold,fill=log10(qval)))+
  scale_fill_continuous(low="#66bb66",high="#bb6666",guide = "colourbar",name= "Log10 q-value")+
  geom_bar(stat="identity")+theme_bw()+
  facet_grid(element~segment,labeller = as_labeller(elements_names))+
  ggtitle("Enrichment of lincRNAs and pc genes in CTCF/cohesin binding sites")+
  xlab("Workspace") + ylab("Fold enrichment")+theme(axis.text.x=element_text(angle = -90, hjust = 0))
#grid.arrange(bs2g,g2bs)

```

##Segments overlap

###Analyses using TAD bins

```{r TAD_bins_enrich_seg, echo=F, warning=F, fig.height=9,fig.width=9}

for(w in c("wholegenome","intergenic","allpc","lincRNA")){
  b2g <- ggplot(data=bins2gene_seg[bins2gene_seg$workspace==w,],aes(x=track,y=fold,fill=log10(qval)))+
    scale_fill_continuous(low="#66bb66",high="#bb6666",guide = "colourbar",name= "Log10\nq-value")+
    geom_bar(stat="identity")+theme_bw()+
    facet_grid(element~annotation,labeller = as_labeller(elements_names))+
    ggtitle(paste0(nice_workspace[w],": \nEnrichment of TAD bins in lincRNAs and pcgenes"))+
    xlab("TAD bins") + ylab("Fold enrichment")+
    theme(axis.text = element_text(size = rel(0.4)))
  
  g2b <- ggplot(data=gene2bins_seg[gene2bins_seg$workspace==w,],aes(x=annottrack,y=fold,fill=log10(qval)))+
    scale_fill_continuous(low="#66bb66",high="#bb6666",guide = "colourbar",name= "Log10\nq-value")+
    geom_bar(stat="identity")+theme_bw()+
    facet_grid(element~segment,labeller = as_labeller(elements_names))+
    ggtitle(paste0(nice_workspace[w],": \nEnrichment of lincRNAs and pc genes in TAD bins"))+
    xlab("TAD bins") + ylab("Fold enrichment")+
    theme(axis.text = element_text(size = rel(0.4)))
  grid.arrange(b2g,g2b)
}

```

###Analyses using CTCF/Cohesin binding sites

```{r bindingsite_enrich_seg, echo=F, warning=F, fig.height=7,fig.width=9}

ggplot(data=bs2gene_seg,aes(x=workspace,y=fold,fill=log10(qval)))+
  scale_fill_continuous(low="#66bb66",high="#bb6666",guide = "colourbar",name= "Log10 q-value")+
  geom_bar(stat="identity")+theme_bw()+
  facet_grid(element~annotation,labeller = as_labeller(elements_names))+
  ggtitle("Enrichment of CTCF/cohesin binding sites in lincRNAs and pcgenes")+
  xlab("Workspace") + ylab("Fold enrichment")+theme(axis.text.x=element_text(angle = -90, hjust = 0))

ggplot(data=gene2bs_seg,aes(x=workspace,y=fold,fill=log10(qval)))+
  scale_fill_continuous(low="#66bb66",high="#bb6666",guide = "colourbar",name= "Log10 q-value")+
  geom_bar(stat="identity")+theme_bw()+
  facet_grid(element~segment,labeller = as_labeller(elements_names))+
  ggtitle("Enrichment of lincRNAs and pc genes in CTCF/cohesin binding sites")+
  xlab("Workspace") + ylab("Fold enrichment")+theme(axis.text.x=element_text(angle = -90, hjust = 0))
#grid.arrange(bs2g,g2bs)

```

