---
title: 'Report 7B: GAT tweaks'
output: pdf_document
author: "Cyril Matthey-Doret"
date: "4 November 2016"
---

```{r setup, echo=F}

#setwd("/home/cyril/Documents/First_step/data/")
#setwd("/home/cyril/Documents/Master/sem_1/First_step/data/")
setwd("/Users/cmatthe5/Documents/First_step/data/")
whole_gat_seg <- read.table("GAT/out/whole_seg_10kgat_fullover_results.txt",header=T)
sets <- c("e.np","ne.p","ne.np")
nice_workspace <- c(wholegenome= "Whole genome",intergenic="Intergenic space",allpc="Protein-coding space", lincRNA="Expressed lincRNAs space")


gene2bins_seg<-droplevels(whole_gat_seg[whole_gat_seg$annotation=="short10bins" & whole_gat_seg$segment %in% sets,])
gene2bins_seg$annottrack <- as.character(gene2bins_seg$annottrack)
gene2bins_seg$annottrack <- factor(gene2bins_seg$annottrack,levels = c(1:20))

CTCF_gene_seg<-droplevels(whole_gat_seg[whole_gat_seg$annotation %in% sets & whole_gat_seg$segment=="GM12878CTCF",])
SMC3_gene_seg<-droplevels(whole_gat_seg[whole_gat_seg$annotation %in% sets & whole_gat_seg$segment=="GM12878SMC3",])
RAD21_gene_seg<-droplevels(whole_gat_seg[whole_gat_seg$annotation %in% sets & whole_gat_seg$segment=="GM12878RAD21",])

library(ggplot2);library(gridExtra)
elements_names <- c(
  `pr` = "promoter",
  `prb` = "promoter + gene body",
  `ne.np` = "non-enhancer,non-promoter",
  `ne.p` = "non-enhancer, promoter",
  `e.np` = "enhancer, non-promoter"
)

```

##Introduction
Here I increased the sampling of GAT to 10000 instead of 1000 to give it more power. I also worked with 10 bins per TAD instead of 20. I only used segment overlap as a measure, not nucleotide. When using expressed lincRNA, intergenic space and all protein-coding genes as workspace, I had to change the --nbuckets parameter from the default of 100000 to 270000 so that nbuckets*bucket size was larger than my largest segment of 269000. This also slows down the computations. Finally, I split the binding sites for CTCF, SCM3 and RAD21 into 3 different analysis so that if the results are only positive for 1 of the proteins, it will not be hidden.

This is a version of report 7 in which genes are categorized by the combination of their promoter/enhancer overlap status.
nep = overlap promoter only (non-enhancer; promoter), enp = overlap enhancer only (enhancer;non-promoter), nenp = overlap none (non-enhancer;non-promoter).

#Results

#Analysis using TAD bins

```{r TAD_bins_enrich, echo=F, results="asis", warning=F, fig.height=9,fig.width=9}

for(w in c("intergenic","allpc")){
  gset <- ifelse(w=="intergenic",yes = "lincRNA ",no="PCG")
  p<- ggplot(data=gene2bins_seg[gene2bins_seg$workspace==w,],aes(x=annottrack,y=fold,fill=log10(qval)))+
    scale_fill_continuous(low="#66bb66",high="#bb6666",guide = "colourbar",name= "Log10\nq-value")+
    geom_bar(stat="identity")+theme_bw()+
    facet_grid(element~segment,labeller = as_labeller(elements_names))+
    ggtitle(paste0(nice_workspace[w],": \nEnrichment of ",gset," in TAD bins"))+
    xlab("TAD bins") + ylab("Fold enrichment")+
    theme(axis.text = element_text(size = rel(0.4)))
  grid.arrange(p)
}

```


#Analysis using CTCF binding sites

```{r bindingsite_CTCF, echo=F, warning=F, fig.height=6,fig.width=9}

ggplot(data=CTCF_gene_seg,aes(x=workspace,y=fold,fill=log10(qval)))+
  scale_fill_continuous(low="#66bb66",high="#bb6666",guide = "colourbar",name= "Log10 q-value")+
  geom_bar(stat="identity")+theme_bw()+
  facet_grid(element~annotation,labeller = as_labeller(elements_names))+
  ggtitle("Enrichment of CTCF binding sites in lincRNAs and pcgenes")+
  xlab("Workspace") + ylab("Fold enrichment")+theme(axis.text.x=element_text(angle = -90, hjust = 0))

#grid.arrange(CTCF_g,g_CTCF)

```

#Analysis using RAD21 binding sites

```{r bindingsite_RAD21, echo=F, warning=F, fig.height=6,fig.width=9}

ggplot(data=RAD21_gene_seg,aes(x=workspace,y=fold,fill=log10(qval)))+
  scale_fill_continuous(low="#66bb66",high="#bb6666",guide = "colourbar",name= "Log10 q-value")+
  geom_bar(stat="identity")+theme_bw()+
  facet_grid(element~annotation,labeller = as_labeller(elements_names))+
  ggtitle("Enrichment of RAD21 binding sites in lincRNAs and pcgenes")+
  xlab("Workspace") + ylab("Fold enrichment")+theme(axis.text.x=element_text(angle = -90, hjust = 0))

#grid.arrange(RAD21_g,g_RAD21)

```

#Analysis using SMC3 binding sites

```{r bindingsite_SMC3, echo=F, warning=F, fig.height=6,fig.width=9}

ggplot(data=SMC3_gene_seg,aes(x=workspace,y=fold,fill=log10(qval)))+
  scale_fill_continuous(low="#66bb66",high="#bb6666",guide = "colourbar",name= "Log10 q-value")+
  geom_bar(stat="identity")+theme_bw()+
  facet_grid(element~annotation,labeller = as_labeller(elements_names))+
  ggtitle("Enrichment of SMC3 binding sites in lincRNAs and pcgenes")+
  xlab("Workspace") + ylab("Fold enrichment")+theme(axis.text.x=element_text(angle = -90, hjust = 0))

#grid.arrange(SMC3_g,g_SMC3)

```

#Conclusions

After splitting genes into these categories, it appears that enhancer-bound genes (both lincRNAs and PCGs) are not enriched at TAD boundaries. This is surprising.

Also, enhancer-bound lincRNAs seem to be more enriched than enhancer-bound protein-coding genes in cohesin, but not in CTCF binding sites for which they both have a ~5 fold enrichment.